RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc

```diff
@@ -0,0 +1,102 @@
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Utilities/interface/do_nothing_deleter.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "CondFormats/DataRecord/interface/BeamSpotObjectsRcd.h"
+//#include "CondFormats/BeamSpotObjects/interface/BeamSpotObjects.h"
+#include "CondFormats/BeamSpotObjects/interface/BeamSpotOnlineObjects.h"
+#include "CondFormats/DataRecord/interface/BeamSpotOnlineLegacyObjectsRcd.h"
+#include "CondFormats/DataRecord/interface/BeamSpotOnlineHLTObjectsRcd.h"
+#include "CondFormats/DataRecord/interface/BeamSpotTransientObjectsRcd.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+#include "FWCore/Framework/interface/ESProductHost.h"
+#include "FWCore/Utilities/interface/ReusableObjectHolder.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include <memory>
+#include <iostream>
+#include <string>
+
+using namespace edm;
+
+class OnlineBeamSpotESProducer : public edm::ESProducer {
+public:
+  OnlineBeamSpotESProducer(const edm::ParameterSet& p);
+  ~OnlineBeamSpotESProducer() override;
+  std::shared_ptr<const BeamSpotObjects> produce(const BeamSpotTransientObjectsRcd&);
+  static void fillDescriptions(edm::ConfigurationDescriptions& desc);
+
+private:
+  const BeamSpotOnlineObjects* compareBS(const BeamSpotOnlineObjects* bs1, const BeamSpotOnlineObjects* bs2);
+  BeamSpotObjects* fakeBS_;
```

Comment:
Might be better to use a smart pointer such as `std::unique_ptr<BeamSpotObject>` here for safety reasons [1].

[1] http://cms-sw.github.io/cms_coding_rules.html#2--naming-rules-1 7.7
> For ownership of dynamic memory, don’t use bare pointers but rather smart pointers: std::unique_ptr, std::shared_ptr, and std::weak_ptr and their constructors std::make_unique<T>() and std::make_shared<T>() (smart pointers).

Reply:
or possible `shared_ptr` per the logic below?

Reply:
It could be a regular (i.e. non-pointer) member as well.

Reply:
I went for shared_ptr following the logic below.

Reply:
I am fine with any solution, in the commit I am going to push soon (after the test) I am using a shared_ptr and kept the &(*fakeBS_) as for the error message (see other comment). If you end up with a different suggestion I can re-iterate with another commit.